src/app/main/login/login.component.ts

```diff
@@ -27,15 +28,26 @@
   }
 
   login(){
+    const HTTP_OPTIONS = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/x-www-form-urlencoded',
+      })
+    };
+
+    const body = new HttpParams()
+      .set('username', this.form.value.username)
+      .set('password', this.form.value.password)
+      .set('grant_type', 'password');
     let b = this.form.value
     console.log(b)
-    this._api.postTypeRequest('login', b).subscribe((res: any) => {
+    this._api.postTypeRequest('login', body, HTTP_OPTIONS).subscribe((res: any) => {
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
    this._api.postTypeRequest("login", body, HTTP_OPTIONS).subscribe((res: any) => {
```